### PR TITLE
change the default config namespace to istio-system

### PR DESCRIPTION
**Pull request by jmuk**
**Release note**:

```release-note
none
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1402)

### CODE REVIEW

File: test/e2e/e2e_report_test.go [hunk 1]
```diff
@@ -81,12 +81,12 @@ apiVersion: "config.istio.io/v1alpha2"
...
   name: rule1
-  namespace: istio-config-default
+  namespace: istio-system
 spec:
   selector: target.name == "*"
   actions:
-  - handler: fakeHandlerConfig.fakeHandler.istio-config-default
-    instances: [ reportInstance.samplereport.istio-config-default ]
+  - handler: fakeHandlerConfig.fakeHandler.istio-system
+    instances: [ reportInstance.samplereport.istio-system ]
```
Comment (douglas-reid) [inline test/e2e/e2e_report_test.go]: @mandarjog: while we are changing this, should we adopt the style of:

```
instances:
- reportInstance...
```
instead?

Comment (mandarjog) [reply test/e2e/e2e_report_test.go]: good point. yes, we should.

@jmuk 
1. use the extended yaml format.
2. elide namespace from handler and instance names.

Comment (jmuk) [reply test/e2e/e2e_report_test.go]: done here. Regarding 2, do we want to elide namespaces in other parts, especially in `testdata/config` files?

**Comment by mandarjog** [comment]
This will work now that mixer builds work with Basel build 0.6.1 now.
/lgtm

**Pull request merged**